buildbot/check.py

```diff
@@ -23,9 +23,47 @@ def do_check(args):
     ret = True
     return ret
 
+def do_check_sycl_unittests(args):
+    ret = False
+
+    env_tmp=os.environ
+    env_tmp["GTEST_FILTER"] = args.gtest_filter
+
+    pitests = os.path.join("tools", "sycl", "unittests", "pi", "PiTests")
+    unittests_cmd = [pitests]
```

Comment:
Why can't we launch these unit-tests using the existing `do_check` function? As I can see, it launches `ninja` and passes a target there - the only thing we need is to pass different target (or several targets)

Reply:
From what I see none of the different LLVM projects creates a target for their unittests. I mirrored that so far.

That does not stop us from introducing a target though. Should I add one?

Reply:
As I can see, the core LLVM & clangs have such targets:
```
$ make help | grep unit
... check-clang-cxx-module-module.unit
... check-clang-unit
... check-llvm-unit
```

I think having such target greatly simplifies unit-tests launch: you don't need to know a path to a test binary or its anymore, the only thing is needed is to launch a special target.

Of course you should know name of this target, but we could use something like `check-sycl-unit` which probably can be easily found in the list of all targets available.

Reply:
For example `check-llvm-unit` and `check-clang-unit` are LIT tests and their names are auto-generated from LIT sub-directory names. It seems that LLVM has targets for LIT but not for unittests (I wonder if that is due to the target name conflicts that could lead to).

We could use a naming scheme like `check-unittest-sycl-...` as that does not seem to be used inside LLVM (yet).

Reply:
> For example `check-llvm-unit` and `check-clang-unit` are LIT tests and their names are auto-generated from LIT sub-directory names. 

Actually, these targets are a little bit tricky: as you can see [here](https://github.com/intel/llvm/tree/sycl/clang/test/Unit), `Unit` sub-directory contains only `lit.cfg` and `lit.site.cfg` to point to actual unit-tests

>  It seems that LLVM has targets for LIT but not for unittests (I wonder if that is due to the target name conflicts that could lead to).

Yes, it seems so. Only generic targets for launching all unit-tests are provided.

I don't know how important is to have mechanism for selecting particular unit-test via target, but it could be definitely useful for developers.

> We could use a naming scheme like `check-unittest-sycl-...` as that does not seem to be used inside LLVM (yet).

@bjoernknafla, what do you think about aligning with LLVM here? I mean `check-sycl-unit` as umbrella target for all unit-tests and `check-sycl-unit-...` as separate targets for specific unit-tests?

Reply:
Interesting find about the LIT config for googletests! I have to dig a bit deeper to fully understand how googletest unittests and LIT interact as my understanding so far was that there are only pure-LIT unittests but I clearly don't have a complete view of this yet.

Reply:
Thank you for being so persistent to align this with the rest of LLVM. The directory you pointed out indeed has a special LIT config in it to understand how to run the tests in the `unittests` directory.
Adapting this as follows:
* Our previous LIT tests now use "SYCL" as their `config.name`while
* LIT untitests use `SYCL-Unit`.

This change is aligned with how Clang does it.

No need for extra targets or most changes to the buildbot scripts (investigating).